1.fee_bumping/fee_bumping.md

```diff
@@ -153,25 +153,217 @@ Bitcoin blockchain, such as exchanges or custodians:
+  replacement transaction in future, it must signal opt-in RBF on one of the
+  inputs of the replacement transaction.
+
+> 1. The original transactions signal replaceability explicitly or through
+> inheritance as described in the above Summary section.
+
+- implication 1: It is not sufficient to look at an individual transaction
+  to determine whether it can be replaced under this policy. A transaction
+  may be replaceable because one of its unconfirmed ancestors signals
+  opt-in RBF.
+- implication 2: A transaction's replaceability may change! If it is does
+  not signal opt-in replaceability, but one of its ancestors does, then it
+  is replaceable. If the ancestor then gets confirmed, the descendant
+  transaction becomes non-replaceable.
+
+> 2. The replacement transaction does not contain any new unconfirmed inputs
+> that did not previously appear in the mempool. (Unconfirmed inputs are
+> inputs spending outputs from currently unconfirmed transactions.)
+
+Rationale: this is an anti-DoS measure to stop a spender creating a replacement
+transaction which has a higher feerate than the original transaction, but
+adds new ancestor dependencies so that the package feerate for the transaction
+is lower.
+
+- implication 1: if a wallet runs out of confirmed transactions (ie all of its
+  coins are tied up in unconfirmed transactions), then it can't use the output
+  of some of those transactions as additional inputs to bump the fee on its other
+  transactions.
 
-### 5 conditions for RBF
-
-TODO
+> 3. The replacement transaction pays an absolute fee of at least the sum paid
+> by the original transactions.
+
+- implication 1: if outputs from the original transaction have been spent, then
+  the replacement transaction must exceed both the *feerate* and *total fee* of
+  the original transaction and its descendants. If the child transaction is large,
+  or there are many descendant transactions, then the total fee required to
+  replace the original transaction could be very large. This is known as 
+  *transaction pinning*.
+
+> 4. The replacement transaction must also pay for its own bandwidth at or
+> above the rate set by the node's minimum relay fee setting. For example,
+> if the minimum relay fee is 1 satoshi/byte and the replacement transaction
+> is 500 bytes total, then the replacement must pay a fee at least 500
+> satoshis higher than the sum of the originals.
+
+Rationale: this is an anti-DoS measure to prevent an adversary from repeatedly
+sending replacement transactions with miniscule feebumps and making the peer
+revalidate many versions of the same transaction.
+
+- implication 1: for transactions at low feerates, the proportional change
+  in feerate from bumping is large. A transaction with a feerate of
+  1 sat/byte needs to *double* its fee to satisfy this rule. For very
+  large transactions (eg large consolidations), then the amount of money
+  spent on fees could be high.
```

Comment:
It might be worthwhile to note that Bitcoin Core (and maybe other nodes) now uses a different setting for this parameter than the minimum fee rate so that these two settings can be configured differently.  The new setting is "incremental relay fee".  E.g.,

```
$ bitcoind -help-debug | grep incre -A3
  -incrementalrelayfee=<amt>
       Fee rate (in BTC/kB) used to define cost of relay, used for mempool
       limiting and BIP 125 replacement. (default: 0.00001)
```

Reply:
added